tee-worker/omni-executor/aa-contracts/aa-demo-app/src/components/TEETokenTransfer.tsx

```diff
@@ -0,0 +1,352 @@
+					balance: BigInt(0),
+					decimals: token.decimals,
+					address: token.address,
+				});
+			}
+		}
+		setTokenBalances(balances);
+	};
+
+	// Fetch current nonce
+	const fetchNonce = async () => {
+		if (!omniAccountAddress || !publicClient) return;
+
+		try {
+			const currentNonce = (await publicClient.readContract({
+				address: CONTRACTS.EntryPoint.address,
+				abi: CONTRACTS.EntryPoint.abi,
+				functionName: "getNonce",
+				args: [omniAccountAddress as `0x${string}`, BigInt(0)],
+			})) as bigint;
+			setNonce(currentNonce);
+		} catch (error) {
+			console.error("Error fetching nonce:", error);
+		}
+	};
+
+	useEffect(() => {
+		fetchBalances();
+		fetchNonce();
+		const interval = setInterval(() => {
+			fetchBalances();
+			fetchNonce();
+		}, 10000);
+		return () => clearInterval(interval);
+	}, [omniAccountAddress, publicClient]);
+
+	const handleSubmit = async (e: React.FormEvent) => {
+		e.preventDefault();
+		setError(null);
+		setTxHash(null);
+
+		// Validation
+		if (!isAddress(recipient)) {
+			setError("Invalid recipient address");
+			return;
+		}
+
+		if (!amount || parseFloat(amount) <= 0) {
+			setError("Invalid amount");
+			return;
+		}
+
+		const token = selectedToken === "USDC" ? ERC20_TOKENS.USDC : ERC20_TOKENS.USDT;
+		const tokenBalance = tokenBalances.find(tb => tb.symbol === selectedToken);
+		
+		if (!tokenBalance) {
+			setError("Token balance not loaded");
+			return;
+		}
+
+		const amountBigInt = parseUnits(amount, token.decimals);
+		
+		if (amountBigInt > tokenBalance.balance) {
+			setError("Insufficient token balance");
+			return;
+		}
+
+		setIsSubmitting(true);
+
+		try {
+			// Build the UserOperation for token transfer
+			const userOp = buildTokenTransferUserOp({
+				omniAccountAddress: omniAccountAddress as `0x${string}`,
+				tokenAddress: token.address,
+				recipient: recipient as `0x${string}`,
+				amount: amountBigInt,
+				nonce,
+			});
+
+			// Pack the UserOperation
+			const packedOp = packUserOperation(userOp);
+			
+			// Convert to serializable format
+			const serializableOp = toSerializablePackedUserOperation(packedOp);
+
+			console.log("Submitting UserOperation through TEE worker:", serializableOp);
+
+			// Submit through TEE worker
+			const response = await submitUserOpTest(
+				[serializableOp],
+				chainId,
+				0, // wallet_index
+				omniAccountHash,
+				DEFAULT_CLIENT_ID
+			);
+
+			console.log("TEE Worker response:", response);
+
+			if (response.transaction_hash) {
+				setTxHash(response.transaction_hash);
+				// Clear form
+				setRecipient("");
+				setAmount("");
+				// Refresh balances and nonce
+				fetchBalances();
+				fetchNonce();
+			} else {
+				setError("Transaction failed - no hash returned");
+			}
+		} catch (err: any) {
+			console.error("Error submitting transaction:", err);
+			setError(err.message || "Failed to submit transaction");
+		} finally {
+			setIsSubmitting(false);
+		}
+	};
+
+	if (!isDeployed) {
```

Comment:
This should be possible in one step (`init_code` + `call_data` together in first UserOp).